### PR TITLE
Settable radar ranges for all ShipTemplateBasedObjects

### PR DESCRIPTION
**Pull request by oznogon**
- Move short- and long-range radar range values, getter, and setter from PlayerSpaceship to ShipTemplateBasedObject.
- Update export lines to add radar range modifications.
- Move GM radar range tweaks from Player2 tab to Ship tab.
- Use short-range radar range for friendly shared radar reveals on Relay.

Example scenario:

```
CpuShip():setTemplate("Atlantis X23"):setFaction("Human Navy"):setPosition(10000, 0):setShortRangeRadarRange(10000)
SpaceStation():setTemplate("Huge Station"):setFaction("Human Navy"):setPosition(-10000, 0):setShortRangeRadarRange(20000)
PlayerSpaceship():setTemplate("Atlantis"):setFaction("Human Navy"):setPosition(0, 0):setLongRangeRadarRange(30000)
```

- The CpuShip has a short-range radar range of 10U, shared with the Player on Relay.
- The SpaceStation has a short-range radar range of 20U, shared with the Player on Relay.
- The PlayerSpaceship has a long-range radar range of 30U, visible from its Science screen.

### CODE REVIEW

**Comment by oznogon** [comment]
Currently there is no value to having long_range_radar_range on non-PlayerSpaceships, but AI states could use both it and short_range_radar_range for enemy and target position ranges, rathern than hard-coded values of 6U/7U/30U.

File: src/screens/crew6/relayScreen.cpp [hunk 1]
```diff
@@ -218,15 +218,15 @@ void RelayScreen::onDraw(sf::RenderTarget& window)
         bool near_friendly = false;
         foreach(SpaceObject, obj, space_object_list)
         {
-            if ((!P<SpaceShip>(obj) && !P<SpaceStation>(obj)) || !obj->isFriendly(my_spaceship))
```
Comment (daid) [inline src/screens/crew6/relayScreen.cpp:221]: For readability and performance, it's better to get the P<ShipTemplateBasedObject> cast once and store it in a local variable.

File: src/spaceObjects/playerSpaceship.cpp [hunk 2]
```diff
@@ -2004,16 +1994,16 @@ void PlayerSpaceship::drawOnGMRadar(sf::RenderTarget& window, sf::Vector2f posit
     SpaceShip::drawOnGMRadar(window, position, scale, rotation, long_range);
     if (long_range)
     {
-        sf::CircleShape radar_radius(long_range_radar_range * scale);
-        radar_radius.setOrigin(long_range_radar_range * scale, long_range_radar_range * scale);
+        sf::CircleShape radar_radius(getShortRangeRadarRange() * scale);
```
Comment (daid) [inline src/spaceObjects/playerSpaceship.cpp:1997]: long has been changed to short incorrectly.

**Review by daid — COMMENTED** [review]
Overall, looks good. Few small things. Idea to use long range radar for AI isn't bad as well.

(github also reports a conflict with master)

**Comment by oznogon** [comment]
@daid Fixed up the issues and resolved merge conflicts. If this gets merged, I'll follow up with the AI changes.

File: src/screenComponents/radarView.cpp [hunk 3]
```diff
@@ -600,27 +608,34 @@ void GuiRadarView::drawObjects(sf::RenderTarget& window_normal, sf::RenderTarget
...
             {
                 P<ScanProbe> sp = obj;
+
                 if (!sp || sp->owner_id != my_spaceship->getMultiplayerId())
                 {
                     continue;
                 }
             }
 
+            float r = (stb_obj && obj->isFriendly(my_spaceship)) ? stb_obj->getShortRangeRadarRange() : 5000.0f;
```
Comment (daid) [inline src/screenComponents/radarView.cpp:630]: Not sure why the `obj->isFriendly(my_spaceship)` check is in here?

File: src/screens/crew6/relayScreen.cpp [hunk 4]
```diff
@@ -222,16 +222,20 @@ void RelayScreen::onDraw(sf::RenderTarget& window)
...
                 P<ScanProbe> sp = obj;
+
                 if (!sp || sp->owner_id != my_spaceship->getMultiplayerId())
                 {
                     continue;
                 }
             }
-            if (obj->getPosition() - target->getPosition() < 5000.0f)
+
+            if (obj->getPosition() - target->getPosition() < stb_obj->getShortRangeRadarRange())
```
Comment (daid) [inline src/screens/crew6/relayScreen.cpp:238]: isn't it possible for stb_obj to be nullptr here for a probe?

File: src/screenComponents/radarView.cpp (hunk 3 above)
Comment (oznogon) [reply src/screenComponents/radarView.cpp]: It's redundant; removed it

File: src/screens/crew6/relayScreen.cpp (hunk 4 above)
Comment (oznogon) [reply src/screens/crew6/relayScreen.cpp]: Yep. Rewrote this check to align with GuiRadarView's

**Pull request merged**